agent/consul/leader_connect_ca_test.go

```diff
@@ -693,6 +693,59 @@ func TestCAManager_Initialize_Vault_WithIntermediateAsPrimaryCA(t *testing.T) {
 	})
 }
 
+func TestCAManager_Verify_Vault_NoChangeToSecondaryConfig(t *testing.T) {
+	if testing.Short() {
+		t.Skip("too slow for testing.Short")
+	}
+	ca.SkipIfVaultNotPresent(t)
+
+	vault := ca.NewTestVaultServer(t)
+
+	_, sDC1 := testServerWithConfig(t, func(c *Config) {
+		c.CAConfig = &structs.CAConfiguration{
+			Provider: "vault",
+			Config: map[string]interface{}{
+				"Address":             vault.Addr,
+				"Token":               vault.RootToken,
+				"RootPKIPath":         "pki-root/",
+				"IntermediatePKIPath": "pki-intermediate/",
+			},
+		}
+	})
+	defer sDC1.Shutdown()
+	testrpc.WaitForActiveCARoot(t, sDC1.RPC, "dc1", nil)
+
+	_, sDC2 := testServerWithConfig(t, func(c *Config) {
+		c.Datacenter = "dc2"
+		c.PrimaryDatacenter = "dc1"
+		c.CAConfig = &structs.CAConfiguration{
+			Provider: "vault",
+			Config: map[string]interface{}{
+				"Address":             vault.Addr,
+				"Token":               vault.RootToken,
+				"RootPKIPath":         "pki-root/",
+				"IntermediatePKIPath": "pki-intermediate/",
+			},
+		}
+	})
+	defer sDC2.Shutdown()
+	joinWAN(t, sDC2, sDC1)
+	testrpc.WaitForActiveCARoot(t, sDC2.RPC, "dc2", nil)
+
+	codec := rpcClient(t, sDC2)
+	var configBefore structs.CAConfiguration
+	err := msgpackrpc.CallWithCodec(codec, "ConnectCA.ConfigurationGet", &structs.DCSpecificRequest{}, &configBefore)
+	require.NoError(t, err)
+
+	renewLeafSigningCert(t, sDC1.caManager, sDC1.caManager.primaryRenewIntermediate)
```

Comment:
I tried this test out on `main` (without the fix) and I saw it passing. So I added this small sleep to get it to fail. I generally try to avoid any kind of sleeping in a test, but in this case I don't see any way to avoid it. We need to give the secondary a few milliseconds to notice the update before we query it again.

The test fails reliably for me with this 100ms sleep, both on main and on this branch. The test could flake, but a flake would be when the test passes instead of fails. That should be less of a concern because it reliably fails in most cases, and won't ever incorrectly fail.

```suggestion
	renewLeafSigningCert(t, sDC1.caManager, sDC1.caManager.primaryRenewIntermediate)
	
	// Give the secondary some time to notice the update
	time.Sleep(100 * time.Millisecond)	
```